apple/testing/default_runner/ios_test_runner.bzl

```diff
@@ -21,9 +21,11 @@ load(
 
 def _get_template_substitutions(ctx):
     """Returns the template substitutions for this runner."""
+    os_version = ctx.attr.os_version or ctx.fragments.objc.ios_simulator_version or ""
+    device_type = ctx.attr.device_type or ctx.fragments.objc.ios_simulator_device or ""
     subs = {
-        "device_type": ctx.attr.device_type,
-        "os_version": ctx.attr.os_version,
+        "device_type": device_type,
+        "os_version": str(os_version),
```

Comment:
Why does this one need `str()`?

Reply:
Otherwise the value from the flag is a DottedVersion

Reply:
Yes the None case is why I'm doing it this way